<a href="https://colab.research.google.com/github/pylabview/Project2_Heart_Failure_Prediction_Dataset/blob/main/Project_2_DS_Sales_Predicctions_Final.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

##Data Science Heart_Failure_Prediction_Dataset
Rodrigo Arguello-Serrano

>This first project will be a sales prediction for food items sold at various stores. The goal of this is to help the retailer understand the properties of products and outlets that play crucial roles in increasing sales.

<a name="TOC"></a> 
#🗂  Table of Contents
1. [🖌 Project 1 - Parts 1 & 2: Source Control and Data Cleaning](#1)
  - [📒 Data Dictionary ](#1.1)
  - [Module Imports & Mounting Goolgle Drive](#1.2)
  - [Loading Data](#1.3)
  - [🧽 Data Cleaning](#1.4)





<a name="1"></a>
#Project 1 - Parts 1 & 2: Source Control and Data Cleaning

⬅️[Go back to TOC](#TOC)

Part 1: Seeting GitHub Repo

[Project GitHub Repo](https://github.com/pylabview/Project2_Heart_Failure_Prediction_Dataset)

<a name="1.2"></a>
## Module Imports & Mounting Goolgle Drive

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from matplotlib.ticker import FuncFormatter
from google.colab import drive
#Setting plots styles
plt.style.use(['dark_background','seaborn-muted', 'seaborn-poster'])
#Moubt Gogle Drive
drive.mount('/content/drive')

Mounted at /content/drive


<a name="1.3"></a>
## Loading Data

In [2]:
df = pd.read_csv("/content/drive/MyDrive/Colab Notebooks/Assignments/Data/heart.csv")
df.head()

Age Sex ChestPainType  RestingBP  Cholesterol  FastingBS RestingECG  MaxHR  \
0   40   M           ATA        140          289          0     Normal    172   
1   49   F           NAP        160          180          0     Normal    156   
2   37   M           ATA        130          283          0         ST     98   
3   48   F           ASY        138          214          0     Normal    108   
4   54   M           NAP        150          195          0     Normal    122   

  ExerciseAngina  Oldpeak ST_Slope  HeartDisease  
0              N      0.0       Up             0  
1              N      1.0     Flat             1  
2              N      0.0       Up             0  
3              Y      1.5     Flat             1  
4              N      0.0       Up             0

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 918 entries, 0 to 917
Data columns (total 12 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Age             918 non-null    int64  
 1   Sex             918 non-null    object 
 2   ChestPainType   918 non-null    object 
 3   RestingBP       918 non-null    int64  
 4   Cholesterol     918 non-null    int64  
 5   FastingBS       918 non-null    int64  
 6   RestingECG      918 non-null    object 
 7   MaxHR           918 non-null    int64  
 8   ExerciseAngina  918 non-null    object 
 9   Oldpeak         918 non-null    float64
 10  ST_Slope        918 non-null    object 
 11  HeartDisease    918 non-null    int64  
dtypes: float64(1), int64(6), object(5)
memory usage: 86.2+ KB


<a name="1.4"></a>
## 🧽 Data Cleaning

In [4]:
# 1) How many rows and columns?
num_cols,num_rows = df.shape
print(f"Number of Columns: {num_cols}, Number or rows: {num_rows}")

Number of Columns: 918, Number or rows: 12


In [5]:
# 2) What are the datatypes of each variable?
df.dtypes

Age                 int64
Sex                object
ChestPainType      object
RestingBP           int64
Cholesterol         int64
FastingBS           int64
RestingECG         object
MaxHR               int64
ExerciseAngina     object
Oldpeak           float64
ST_Slope           object
HeartDisease        int64
dtype: object

In [6]:
# Are there duplicates? If so, drop any duplicates.
dups = df.duplicated().sum()
print(f"Number of duplicates: {dups}")

Number of duplicates: 0


In [7]:
# 4) Identify missing values.
total_missing_series = df.isna().sum()
total_missing_series_idx = df.isna().sum().index
total_cols = len(total_missing_series_idx )

for i in range(total_cols):
  if total_missing_series[i] > 0:
    print(f"Row index {total_missing_series_idx[i]}, number of missing {total_missing_series[i]}")

In [11]:
# 6) Confirm that there are no missing values after addressing them.
print(f"Are All missing values removed? {df.isna().sum().sum()==0}")

Are All missing values removed? True


1. Source of data
>[kaggle - Heart Failure Prediction Dataset](https://www.kaggle.com/datasets/fedesoriano/heart-failure-prediction?resource=download)
2. Brief description of data
>Cardiovascular diseases (CVDs) are the number 1 cause of death globally, taking an estimated 17.9 million lives each year, which accounts for 31% of all deaths worldwide. Four out of 5CVD deaths are due to heart attacks and strokes, and one-third of these deaths occur prematurely in people under 70 years of age. Heart failure is a common event caused by CVDs and this dataset contains 11 features that can be used to predict a possible heart disease.<br>

  >People with cardiovascular disease or who are at high cardiovascular risk (due to the presence of one or more risk factors such as hypertension, diabetes, hyperlipidaemia or already established disease) need early detection and management wherein a machine learning model can be of great help.

3. What is the target?
>HeartDisease feature
4. What does one row represent? (A person?  A business?  An event? A product?)
> A person medical data
5. Is this a classification or regression problem?
> Classification
6. How many features does the data have?
>12
7. How many rows are in the dataset?
>918
8. What, if any, challenges do you foresee in cleaning, exploring, or modeling with this dataset?
> None